apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/BigtableReadRowsRequestManager.cs

```diff
@@ -114,9 +113,14 @@ private RowSet FilterRows()
                         StartKeyIsAlreadyRead(rowRange.StartKeyOpen))
                     || startKeyOneofCase == RowRange.StartKeyOneofCase.None)
                 {
-                    newRange.StartKeyOpen = (ByteString) LastFoundKey;
+                    RowRange newRange = rowRange.Clone();
+                    newRange.StartKeyOpen = LastFoundKey.Value;
+                    newRowSet.RowRanges.Add(newRange);
+                }
+                else
+                {
+                    newRowSet.RowRanges.Add(rowRange);
                 }
-                newRowSet.RowRanges.Add(newRange);
```

Comment:
I think this would be a bit cleaner:
```csharp
RowRange newRange = rowRange;
if (...)
{
    newRange = newRange.Clone();
    newRange.StartKeyOpen = LastFoundKey.Value;
}
newRowSet.RowRanges.Add(newRange);
```

Reply:
Much cleaner. 
Is it OK that `newRange` will very likely be initialized and then re-assigned before being added to the `newRowSet`? Although it is only a reference.

Reply:
I don't see any issue with that